core/src/main/scala/org/apache/spark/scheduler/TaskScheduler.scala

```diff
@@ -54,7 +54,7 @@ private[spark] trait TaskScheduler {
   // Submit a sequence of tasks to run.
   def submitTasks(taskSet: TaskSet): Unit
 
-  // Kill all the tasks in a stage and fail the stage and all the jobs that depend on the stage.
+  // Kill all the tasks in all the stage attempts of the same stage Id
```

Comment:
Please add comment about `mark all the stage attempts as zombie`.

Reply:
I considered to metion zombie here but gave up finally as think that it's only an API and we don't we don't enforce the same functionaltiy for every implementation.

Reply:
Sounds good to me.